src/StyleableBannerWithLink/index.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState } from 'react';
+import { Button, LinkButton, SvgCross } from '@guardian/source-react-components';
+import { useEscapeShortcut, OnCloseClick, CLOSE_BUTTON_ID } from '../bannerCommon/bannerActions';
+import type { TrackClick } from '../utils/tracking';
+import { StyleData, selfServeStyles, cssInjectionCheck } from '../styles/bannerCommon';
+import { canRender, COMPONENT_NAME } from './canRender';
+export { COMPONENT_NAME };
+
+export type BrazeMessageProps = {
+    ophanComponentId?: string;
+    styleBackground?: string;
+    header?: string;
+    styleHeader?: string;
+    body?: string;
+    styleBody?: string;
+    highlight?: string;
+    styleHighlight?: string;
+    styleHighlightBackground?: string;
+    buttonText?: string;
+    buttonUrl?: string;
+    styleButton?: string;
+    styleButtonBackground?: string;
+    styleButtonHover?: string;
+    imageUrl?: string;
+    imageAltText?: string;
+    imagePosition?: string;
+    styleClose?: string;
+    styleCloseBackground?: string;
+    styleCloseHover?: string;
+};
+
+const defaultColors: StyleData = {
+    styleBackground: 'rgb(237, 237, 237)',
+    styleHeader: 'rgb(51, 51, 51)',
+    styleBody: 'rgb(51, 51, 51)',
+    styleHighlight: 'rgb(51, 51, 51)',
+    styleHighlightBackground: 'rgb(237, 237, 237)',
+    styleButton: 'rgb(255, 255, 255)',
+    styleButtonBackground: 'rgb(5, 41, 98)',
+    styleButtonHover: 'rgb(35, 75, 138)',
+    styleClose: 'rgb(5, 41, 98)',
+    styleCloseBackground: 'rgb(237, 237, 237)',
+    styleCloseHover: '#e5e5e5',
+};
+
+export type Props = {
+    brazeMessageProps: BrazeMessageProps;
+    trackClick: TrackClick;
+};
+
+const StyleableBannerWithLink: React.FC<Props> = (props: Props) => {
+    const {
+        brazeMessageProps: {
+            ophanComponentId,
+            styleBackground,
+            header,
+            styleHeader,
+            body,
+            styleBody,
+            highlight,
+            styleHighlight,
+            styleHighlightBackground,
+            buttonText,
+            buttonUrl,
+            styleButton,
+            styleButtonBackground,
+            styleButtonHover,
+            imageUrl,
+            imageAltText = '',
+            imagePosition = 'center',
+            styleClose,
+            styleCloseBackground,
+            styleCloseHover,
+        },
+        trackClick,
+    } = props;
+
+    const styles = selfServeStyles(
```

Comment:
I wonder if we can simplify the interface to `selfServeStyles` a bit so that we don't need to specify defaults twice? Once in the inline call to `cssInjectionCheck` and then again as the second argument. What if `selfServeStyles` called `cssInjectionCheck` itself, and then use the second `defaultColors` arg to know what to default to? That way all of the "defaulting" would be happening in one place, rather than being spread out? What do you think?

Reply:
TS error fixed